Realm/Realm/Realm.cs

```diff
@@ -871,6 +871,69 @@ public RealmObject Find(string className, string primaryKey)
 
         #endregion Quick Find using primary key
 
+        #region Thread Handover
+
+        /// <summary>
+        /// Returns the same object as the one referenced when the <see cref="ThreadSafeReference.Object{T}"/> was first created,
+        /// but resolved for the current Realm for this thread. Returns null if this object was deleted after
+        /// the reference was created.
+        /// </summary>
+        /// <returns>A thread-confined instance of the original <see cref="RealmObject"/> resolved for the current thread.</returns>
+        /// <param name="reference">The thread-safe reference to the thread-confined <see cref="RealmObject"/> to resolve in this <see cref="Realm"/>.</param>
+        /// <typeparam name="T">The type of the object, contained in the reference.</typeparam>
+        public T ResolveReference<T>(ThreadSafeReference.Object<T> reference) where T : RealmObject
+        {
+            var objectPtr = SharedRealmHandle.ResolveReference(reference);
+            if (objectPtr == IntPtr.Zero)
+            {
+                return null;
+            }
+
+            return (T)MakeObject(reference.Metadata, objectPtr);
+        }
+
+        /// <summary>
+        /// Returns the same collection as the one referenced when the <see cref="ThreadSafeReference.List{T}"/> was first created,
+        /// but resolved for the current Realm for this thread.
+        /// </summary>
```

Comment:
Please add a clarifying remark about copying so people don't think it copies a list of pointers.